sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/SessionReceiverManager.cs

```diff
@@ -451,9 +454,13 @@ public override async Task CancelAsync()
                 await _sessionLockRenewalTask.ConfigureAwait(false);
             }
 
+            // We do not dispose _sessionLockCancellationSource here because it is exposed to users via the SessionLockLostAsync
```

Comment:
Nice find.  this seems like it would have been a difficult one to track down.

Reply:
The test pipeline failed last night 🎉